dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/QuayImageRegistry.java

```diff
@@ -121,7 +121,7 @@ public QuayImageRegistry(final Token quayToken) {
             final String imageID = quayTag.getImageId();
             List<Checksum> checksums = new ArrayList<>();
             checksums.add(new Checksum(manifestDigest.split(":")[0], manifestDigest.split(":")[1]));
-            return Collections.singleton(new Image(checksums, repo, tag.getName(), imageID));
+            return Collections.singleton(new Image(checksums, repo, tag.getName(), imageID, tool.getRegistry()));
```

Comment:
Following up on my previous comments, if this always returns a max of one Image (?), then the method should return an Optional<Image> and not a Set.